cpp/util/thread_pool.cc

```diff
@@ -114,7 +146,19 @@ void ThreadPool::Add(const function<void()>& closure) {
 
   {
     lock_guard<mutex> lock(impl_->queue_lock_);
-    impl_->queue_.push(closure);
+    impl_->queue_.emplace(make_tuple(system_clock::now(), closure, nullptr));
+  }
+  impl_->queue_cond_var_.notify_one();
+}
+
+
+void ThreadPool::Delay(const duration<double>& delay, util::Task* task) {
+  CHECK_NOTNULL(task);
+  {
+    lock_guard<mutex> lock(impl_->queue_lock_);
+    impl_->queue_.emplace(make_tuple(
+        system_clock::now() + duration_cast<std::chrono::microseconds>(delay),
```

Comment:
There was the need for a `duration_cast` here? `system_clock` uses an integer duration or something like that?